### PR TITLE
fix: upgrading typescript and semantic release. Removal of typesdoc

### LINKED ISSUE #63: Adjust dev libraries
Upgrade typescript to latest. 
Remove unused dependencies. 
Upgrade semantic-release.
Upgrade eslint

### PR DESCRIPTION
**Pull request by Saeger**
Fixes #63

### CODE REVIEW

**Comment by tgun89** [comment]
@polarn

**Comment by polarn** [comment]
Poor @Saeger, you're trolling him!

**Comment by polarn** [comment]
@juliuscc pretty please, with @tgun89 on top, let it pass! 😆

**Comment by juliuscc** [comment]
You shall not pass Gandalf GIF

**Review by tgun89 — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -53,11 +53,11 @@
   },
   "devDependencies": {
     "@types/jest": "^24.9.1",
-    "@typescript-eslint/eslint-plugin": "^1.13.0",
-    "@typescript-eslint/parser": "^1.13.0",
+    "@typescript-eslint/eslint-plugin": "^5.13.0",
```
Comment (tgun89) [inline package.json:56]: maybe use fixed versions instead

**Comment by tgun89** [comment]
@Saeger since @polarn joined i had to approve

**Pull request merged**